### PR TITLE
Improve concepts and related doc sections

### PR DESCRIPTION
**Pull request by mosabua**
## Description

- Add client
- Update data sources
- Add links to website
- Modernize content
- Update clients overview page
- Fix issues in related client protocol docs

## Additional context and related issues

Also fixes some minor grammar and style issues in client protocol page in separate commit.

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: docs/src/main/sphinx/develop/client-protocol.md [hunk 1]
```diff
@@ -9,50 +9,47 @@ It can also be used to implement your own client, if necessary.
...
 
 ## Overview of query processing
 
 A Trino client request is initiated by an HTTP `POST` to the endpoint
-`/v1/statement`, with a `POST` body consisting of the SQL query string.
-The caller may set various {ref}`client-request-headers`. The headers are
-only required on the initial `POST` request, and not when following the
-`nextUri` links.
+`/v1/statement`, with a `POST` body consisting of the SQL query string. The
+caller may set various [](client-request-headers). The headers are only
```
Comment (m57lyra) [inline docs/src/main/sphinx/develop/client-protocol.md:21]: Square brackets are missing the link text

File: docs/src/main/sphinx/develop/client-protocol.md [hunk 2]
```diff
@@ -9,50 +9,47 @@ It can also be used to implement your own client, if necessary.
...
+required on the initial `POST` request, and not when following the `nextUri`
```
Comment (m57lyra) [inline docs/src/main/sphinx/develop/client-protocol.md:22]: In the initial

File: docs/src/main/sphinx/develop/client-protocol.md [hunk 3]
```diff
@@ -9,50 +9,47 @@ It can also be used to implement your own client, if necessary.
...
+links.
 
-If the client request returns an HTTP 502, 503 or 504, that means there was
-intermittent problem processing request and the client should try again
-in 50-100 milliseconds. Trino does not generate those codes by itself
-but those can be generated by gateways/load balancers in front of Trino.
-Any HTTP status other than 502, 503, 504 or 200 means that query processing
-has failed.
+If the client request returns an HTTP 502, 503, or 504, that means there was
+intermittent problem processing request and the client should try again in
```
Comment (m57lyra) [inline docs/src/main/sphinx/develop/client-protocol.md:26]: an intermittent

File: docs/src/main/sphinx/develop/client-protocol.md [hunk 4]
```diff
@@ -9,50 +9,47 @@ It can also be used to implement your own client, if necessary.
...
+50-100 milliseconds. Trino does not generate those codes by itself but those can
```
Comment (m57lyra) [inline docs/src/main/sphinx/develop/client-protocol.md:27]: 1/ Do we actually write milliseconds out? We don't write seconds out I think?

2/ itself, but

File: docs/src/main/sphinx/develop/client-protocol.md [hunk 5]
```diff
@@ -9,50 +9,47 @@ It can also be used to implement your own client, if necessary.
...
 
 The `/v1/statement` `POST` request returns a JSON document of type
-`QueryResults`, as well as a collection of response headers.  The
-`QueryResults` document contains an `error` field of type
-`QueryError` if the query has failed, and if that object is not present,
-the query succeeded.  Important members of `QueryResults` are documented
-in the following sections.
+`QueryResults`, as well as a collection of response headers. The `QueryResults`
+document contains an `error` field of type `QueryError` if the query has failed,
+and if that object is not present, the query succeeded. Important members of
```
Comment (m57lyra) [inline docs/src/main/sphinx/develop/client-protocol.md:34]: The use of members here is awkward

File: docs/src/main/sphinx/develop/client-protocol.md [hunk 6]
```diff
@@ -9,50 +9,47 @@ It can also be used to implement your own client, if necessary.
...
+a `nextUri` link, the query has completed, either successfully or
+unsuccessfully, and no additional requests need to be made. If the `nextUri`
+link is present in the document, there are more query results to be fetched. The
+client should loop executing a `GET` request to the `nextUri` returned in the
+`QueryResults` response object until `nextUri` is absent from the response.
 
 The `status` field of the JSON document is for human consumption only, and
-provides a hint about the query state.  It can not be used to tell if the
-query is finished.
+provides a hint about the query state. It can not be used to tell if the query
```
Comment (m57lyra) [inline docs/src/main/sphinx/develop/client-protocol.md:51]: cannot is one word

File: docs/src/main/sphinx/overview/concepts.md [hunk 7]
```diff
@@ -81,72 +81,126 @@ for task execution.
...
+
+Clients allow you to connect to Trino, submit SQL queries, and receive the
+results. Clients can access all configured data sources using
+[catalogs](trino-concept-catalog). Clients are full-featured applications or
+libraries and drivers that allow you to connect to any application supporting
+that driver, or even your own custom application or script.
+
+Clients include command line tools, desktop applications, web-based
+applications, and software-as-a-service solutions with features such as
+interactive SQL query authoring with editors or rich user interfaces for
```
Comment (m57lyra) [inline docs/src/main/sphinx/overview/concepts.md:95]: editors, or

File: docs/src/main/sphinx/overview/concepts.md [hunk 8]
```diff
@@ -81,72 +81,126 @@ for task execution.
...
+
+[A comprehensive list with more details for each client is available on the
+Trino website](https://trino.io/ecosystem/client).
+
+Clients can process the returned data from Trino and be used for data pipelines
+across catalogs and from other data sources to Trino catalogs.
+
+From a technical perspective, clients only interact with the
+[coordinator](trino-concept-coordinator) of the Trino cluster and use the
+[](/develop/client-protocol).
```
Comment (m57lyra) [inline docs/src/main/sphinx/overview/concepts.md:107]: Link text brackets are empty

File: docs/src/main/sphinx/overview/concepts.md [hunk 9]
```diff
@@ -81,72 +81,126 @@ for task execution.
...
+
+
+(trino-concept-data-source)=
+## Data source
+
+Trino is a query engine that you can use to query many different data sources.
```
Comment (m57lyra) [inline docs/src/main/sphinx/overview/concepts.md:113]: different data sources, including

File: docs/src/main/sphinx/overview/concepts.md [hunk 10]
```diff
@@ -81,72 +81,126 @@ for task execution.
...
+Trino contains [many built-in connectors](/connector):
+
+* Connectors for data lakes and lakehouses including the [Delta
+  Lake](/connector/delta-lake), [Hive](/connector/hive),
+  [Hudi](/connector/hudi), and [Iceberg](/connector/iceberg) connectors.
+* Connectors for relational database management systems, including the
+  [MySQL](/connector/mysql), [PostgreSQL](/connector/postgresql),
+  [Oracle](/connector/oracle), and [SQL Server](/connector/sqlserver)
+  connectors.
+* Connectors for a variety of other systems, including the
```
Comment (m57lyra) [inline docs/src/main/sphinx/overview/concepts.md:151]: There's an "and" missing from this list

File: docs/src/main/sphinx/overview/concepts.md [hunk 11]
```diff
@@ -81,72 +81,126 @@ for task execution.
...
+  [Cassandra](/connector/cassandra), [ClickHouse](/connector/clickhouse),
+  [OpenSearch](/connector/opensearch), [Pinot](/connector/pinot),
+  [Prometheus](/connector/prometheus), [SingleStore](/connector/singlestore),
+  [Snowflake](/connector/snowflake) connectors.
+* A number of other utility connectors such as the [JMX](/connector/jmx),
+  [System](/connector/system), and [TPCH](/connector/tpch) connectors.
+
+Every catalog uses a specific connector. If you examine a catalog configuration
+file, you see that each contains a mandatory property `connector.name`.
```
Comment (m57lyra) [inline docs/src/main/sphinx/overview/concepts.md:160]: And value

File: docs/src/main/sphinx/overview/concepts.md [hunk 12]
```diff
@@ -81,72 +81,126 @@ for task execution.
...
-Hive data source.
-
-When addressing a table in Trino, the fully-qualified table name is
-always rooted in a catalog. For example, a fully-qualified table name
-of `hive.test_data.test` refers to the `test` table in the
-`test_data` schema in the `hive` catalog.
-
-Catalogs are defined in properties files stored in the Trino
-configuration directory.
+A Trino catalog is a collection of configuration properties to access a specific
```
Comment (m57lyra) [inline docs/src/main/sphinx/overview/concepts.md:165]: properties used to access

File: docs/src/main/sphinx/overview/concepts.md [hunk 13]
```diff
@@ -81,72 +81,126 @@ for task execution.
...
+data source, including the required connector and any other details such as
+credentials and URL. Catalogs are defined in properties files stored in the
+Trino configuration directory. The name of the properties file determines the
+name of the catalog. For example, the properties file `etc/example.properties`
+results in a catalog name `example`.
+
+You can configure and use many catalogs, also using the same connector, to
```
Comment (m57lyra) [inline docs/src/main/sphinx/overview/concepts.md:172]: Leave out the word also

File: docs/src/main/sphinx/overview/concepts.md [hunk 14]
```diff
@@ -81,72 +81,126 @@ for task execution.
...
+access different data sources. For example, if you have two data lakes, you can
+configure two catalogs in a single Trino cluster that both use the Hive
+connector, allowing you to query data from both clusters, even within the same
+SQL query. You can also use a Hive connector for one catalog to access a data
+lake, and use the Iceberg connector for another catalog to access the data
+lakehouse. Or you can configure different catalogs to access different
```
Comment (m57lyra) [inline docs/src/main/sphinx/overview/concepts.md:178]: lakehouse. Or, you

File: docs/src/main/sphinx/overview/concepts.md [hunk 15]
```diff
@@ -81,72 +81,126 @@ for task execution.
...
+PostgreSQL database. The combination of different catalog is determined by your
```
Comment (m57lyra) [inline docs/src/main/sphinx/overview/concepts.md:179]: different catalogs

File: docs/src/main/sphinx/overview/concepts.md [hunk 16]
```diff
@@ -81,72 +81,126 @@ for task execution.
...
+needs to access different data sources only.
+
+A catalog contains one or more schemas, which in turn contain objects such as
+tables, views, or materialized views. When addressing an object like table in
```
Comment (m57lyra) [inline docs/src/main/sphinx/overview/concepts.md:183]: do you mean 

object-like 

Or

objects such as tables

File: docs/src/main/sphinx/overview/concepts.md [hunk 17]
```diff
@@ -81,72 +81,126 @@ for task execution.
...
-Schemas are a way to organize tables. Together, a catalog and schema
-define a set of tables that can be queried. When accessing Hive or a
-relational database such as MySQL with Trino, a schema translates to
-the same concept in the target database. Other types of connectors may
-choose to organize tables into schemas in a way that makes sense for
-the underlying data source.
+Schemas are a way to organize tables. Together, a catalog and schema define a
+set of tables and other objects that can be queried. When accessing Hive or a
+relational database such as MySQL with Trino, a schema translates to the same
+concept in the target database. Other types of connectors may choose to organize
```
Comment (m57lyra) [inline docs/src/main/sphinx/overview/concepts.md:193]: connectors may organize 

(Remove choose to)

File: docs/src/main/sphinx/overview/concepts.md [hunk 18]
```diff
@@ -81,72 +81,126 @@ for task execution.
...
 
 ### Table
 
-A table is a set of unordered rows, which are organized into named columns
-with types. This is the same as in any relational database. The mapping
-from source data to tables is defined by the connector.
+A table is a set of unordered rows, which are organized into named columns with
+[types](/language/types). This is the same as in any relational database. The
+mapping from source data to tables is defined by the connector. Type mapping
+varies across connectors and is documented in the specific connector
```
Comment (m57lyra) [inline docs/src/main/sphinx/overview/concepts.md:201]: Type mapping from sou...by the connector, and varies across

**Review by m57lyra — COMMENTED** [review]
Some stuff to fix, esp missing link texts

File: docs/src/main/sphinx/develop/client-protocol.md (hunk 1 above)
Comment (mosabua) [reply docs/src/main/sphinx/develop/client-protocol.md:21]: Sphinx inserts the title automatically now

File: docs/src/main/sphinx/develop/client-protocol.md (hunk 2 above)
Comment (mosabua) [reply docs/src/main/sphinx/develop/client-protocol.md:22]: updated

File: docs/src/main/sphinx/develop/client-protocol.md (hunk 4 above)
Comment (mosabua) [reply docs/src/main/sphinx/develop/client-protocol.md:27]: Updated

File: docs/src/main/sphinx/develop/client-protocol.md (hunk 5 above)
Comment (mosabua) [reply docs/src/main/sphinx/develop/client-protocol.md:34]: changed to "aspects"

File: docs/src/main/sphinx/overview/concepts.md (hunk 14 above)
Comment (mosabua) [reply docs/src/main/sphinx/overview/concepts.md:178]: updated

File: docs/src/main/sphinx/overview/concepts.md (hunk 15 above)
Comment (mosabua) [reply docs/src/main/sphinx/overview/concepts.md:179]: updated

File: docs/src/main/sphinx/overview/concepts.md (hunk 16 above)
Comment (mosabua) [reply docs/src/main/sphinx/overview/concepts.md:183]: Updated

File: docs/src/main/sphinx/overview/concepts.md (hunk 18 above)
Comment (mosabua) [reply docs/src/main/sphinx/overview/concepts.md:201]: updated

File: docs/src/main/sphinx/overview/concepts.md (hunk 17 above)
Comment (mosabua) [reply docs/src/main/sphinx/overview/concepts.md:193]: Updated

File: docs/src/main/sphinx/develop/client-protocol.md (hunk 3 above)
Comment (mosabua) [reply docs/src/main/sphinx/develop/client-protocol.md:26]: Fixed

File: docs/src/main/sphinx/overview/concepts.md (hunk 9 above)
Comment (mosabua) [reply docs/src/main/sphinx/overview/concepts.md:113]: I prefer to have shorter separate sentences.

**Comment by mosabua** [comment]
All comments addressed @m57lyra

File: docs/src/main/sphinx/client.md [hunk 19]
```diff
@@ -16,5 +17,6 @@ client/cli
 client/jdbc
 ```
 
-In addition, the community provides [numerous other clients](https://trino.io/resources.html) for platforms such as Python, and these
-can in turn be used to connect applications using these platforms.
+In addition, the community provides [numerous other client libraries, drivers,
```
Comment (dain) [inline docs/src/main/sphinx/client.md:20]: We should separate out the clients maintained directly by the Trino community.

Comment (mosabua) [reply docs/src/main/sphinx/client.md:20]: Yes .. thats a separate task for the website though .. ping me when you have time to chat about that. I am not sure how to best do that but have a few ideas.

**Review by raunaqmorarka — APPROVED** [review]

**Comment by mosabua** [comment]
Thank you for the review and approval @raunaqmorarka

**Pull request merged**